test/dailyLimitModule.js

```diff
@@ -158,13 +161,44 @@ contract('DailyLimitModule', function(accounts) {
         assert.equal(await testToken.balances(gnosisSafe.address), 80);
         assert.equal(await testToken.balances(accounts[0]), 20);
 
+
         // Third withdrawal will fail
         await utils.assertRejects(
             dailyLimitModule.executeDailyLimit(testToken.address, accounts[0], 10, {from: accounts[0]}),
             "Daily limit exceeded for ERC20 token"
         )
+
         // Balances didn't change
         assert.equal(await testToken.balances(gnosisSafe.address), 80);
         assert.equal(await testToken.balances(accounts[0]), 20);
-    })
+
+        // Withdrawal should  fail because of ERC20 transfer revert
+        let mockContract = await MockContract.new();
+        let mockToken = MockToken.at(mockContract.address);
+        const method = Web3EthAbi.encodeFunctionSignature('transfer(address,uint256)');
+        await mockContract.givenRevertAny(method);
+        await utils.assertRejects(
+            dailyLimitModule.executeDailyLimit(mockContract.address, accounts[0], 10, {from: accounts[0]}),
+            "ERC20 token reverted transfer"
+        );
+
+
+        // Withdrawal should fail because of ERC20 transfer out of gas
+        mockContract = await MockContract.new();
```

Comment:
it should not be necessary to create a new MockContract for every mock

Reply:
previously mocked methods will be overwritten by new mocking calls

The MockContract uses a map and as long as the method id is the same it will overwrite the expected result

Reply:
I am creating the mockContract only once in the new commit :)